lib/elb/models/LoadBalancerConfig.rb

```diff
@@ -0,0 +1,227 @@
+          @listeners = []
+          if json["listeners"]["includes"]
+            json["listeners"]["includes"].each_pair do |name, vars|
+              @listeners << Loader.listener(name, vars)
+            end
+          end
+          if json["listeners"]["inlines"]
+            json["listeners"]["inlines"].each do |inline|
+              @listeners << ListenerConfig.new(inline)
+            end
+          end
+
+          # Map subnets to their actual subnet description from aws
+          @subnets = (json["subnets"] || []).map do |subnet|
+            full_subnet = EC2::named_subnets[subnet] || EC2::id_subnets[subnet]
+
+            if full_subnet.nil?
+              raise "#{subnet} is not a valid subnet name or id"
+            else
+              full_subnet
+            end
+          end
+
+          # Map backend policies to the Aws::ElasticLoadBalancing::Types::BackendServerDescription
+          # reject any local polices with empty policies array which means it should be deleted from aws
+          @backend_policies = (json["backend-policies"] || []).map do |backend|
+            Aws::ElasticLoadBalancing::Types::BackendServerDescription.new({
+              instance_port: backend["port"],
+              policy_names: backend["policies"]
+            })
+          end.reject { |backend| backend.policy_names.empty? }
+
+          @security_groups = json["security-groups"] || []
+          @internal = json["internal"] || false
+          @tags = json["tags"] || {}
+          @manage_instances = json["manage-instances"] || false
+          @health_check = if json["health-check"] then HealthCheckConfig.new(json["health-check"]) end
+          @cross_zone = json["cross-zone"] || false
+          @access_log = if json["access-log"] then AccessLogConfig.new(json["access-log"]) else false end
+          @connection_draining = json["connection-draining"] || false
+          @idle_timeout = json["idle-timeout"]
+        end
+      end
+
+      # Public: Get the config as a prettified JSON string.
+      #
+      # Returns the JSON string
+      def pretty_json
+        JSON.pretty_generate({
+          "listeners" => {
+            "includes" => {},
+            "inlines" => @listeners.map(&:to_hash)
+          },
+          "subnets" => @subnets,
+          "security-groups" => @security_groups,
+          "internal" => @internal,
+          "tags" => @tags,
+          "manage-instances" => @manage_instances,
+          "health-check" => @health_check.to_hash,
+          "backend-policies" => @backend_policies.map do |backend_policy|
+            {
+              "port" => backend_policy.instance_port,
+              "policies" => backend_policy.policy_names
+            }
+          end,
+          "cross-zone" => @cross_zone,
+          "access-log" => if @access_log then @access_log.to_hash else @access_log end,
+          "connection-draining" => @connection_draining,
+          "idle-timeout" => @idle_timeout,
+        })
```

Comment:
I think in a method like this used only for migration, displaying the nil values is helpful so that when you migrate you know that we aren't ignoring those values, they just don't have values in AWS too.